src/otx/algo/common/utils/nms.py

```diff
@@ -39,7 +39,11 @@ def forward(
             valid_mask = scores > score_threshold
             bboxes, scores = bboxes[valid_mask], scores[valid_mask]
             valid_inds = torch.nonzero(valid_mask, as_tuple=False).squeeze(dim=1)
-        inds = torch_nms(bboxes, scores, iou_threshold)
+        if bboxes.get_device() == -1 and bboxes.dtype == torch.bfloat16:  # torch nms kernel doesn't support bfloat16
+            with torch.autocast(device_type="cpu", dtype=torch.bfloat16, enabled=False):
```

Comment:
Just question, is `torch.autocast` needed because `bboxes` and `scores` were created in autocast-enabled region?

Reply:
I added it to be sure that automatic cast is disabled in nms operations. I refered [this](https://pytorch.org/docs/stable/amp.html#autocasting) when I implemented.